sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -115,46 +141,193 @@ export class BatchingReceiver extends MessageReceiver {
+        true
+      );
+    };
+
+    this._getRemainingWaitTimeInMsFn = (
+      maxWaitTimeInMs: number,
+      maxTimeAfterFirstMessageInMs: number
+    ) => getRemainingWaitTimeInMsFn(maxWaitTimeInMs, maxTimeAfterFirstMessageInMs);
+
+    this.isReceivingMessages = false;
+  }
+
+  private _createServiceBusMessage: (
+    context: Pick<EventContext, "message" | "delivery">
+  ) => ServiceBusMessageImpl;
+
+  private _getRemainingWaitTimeInMsFn: typeof getRemainingWaitTimeInMsFn;
+  private _closeHandler: ((connectionError?: AmqpError | Error) => void) | undefined;
+
+  isReceivingMessages: boolean;
+
+  /**
+   * Receives a set of messages,
+   *
+   * @internal
+   * @ignore
+   */
+  public async receiveMessages(args: ReceiveMessageArgs): Promise<ServiceBusMessageImpl[]> {
+    try {
+      this.isReceivingMessages = true;
+      const receiver = await this._getCurrentReceiver(args.userAbortSignal);
+      return await this._receiveMessagesImpl(receiver, args);
     } finally {
       this.isReceivingMessages = false;
-      this._connectionErrorHandler = undefined;
     }
   }
 
-  private _receiveImpl(
-    maxMessageCount: number,
-    maxWaitTimeInMs: number,
-    maxTimeAfterFirstMessageMs: number,
-    abortSignal?: AbortSignalLike
+  /**
+   * Closes the receiver (optionally with an error), cancelling any current operations.
+   *
+   * @param connectionError An optional error (rhea doesn't always deliver one for certain disconnection events)
+   */
+  close(connectionError?: Error | AmqpError) {
+    if (this._closeHandler) {
+      this._closeHandler(connectionError);
+      this._closeHandler = undefined;
+    }
+  }
+
+  private _receiveMessagesImpl(
+    receiver: MinimalReceiver,
+    args: ReceiveMessageArgs
   ): Promise<ServiceBusMessageImpl[]> {
     const getRemainingWaitTimeInMs = this._getRemainingWaitTimeInMsFn(
-      maxWaitTimeInMs,
-      maxTimeAfterFirstMessageMs
+      args.maxWaitTimeInMs,
+      args.maxTimeAfterFirstMessageInMs
     );
 
     const brokeredMessages: ServiceBusMessageImpl[] = [];
+    const loggingPrefix = `[${receiver.connection.id}|r:${receiver.name}`;
 
-    this.isReceivingMessages = true;
     return new Promise<ServiceBusMessageImpl[]>((resolve, reject) => {
       let totalWaitTimer: NodeJS.Timer | undefined;
 
       // eslint-disable-next-line prefer-const
       let cleanupBeforeResolveOrReject: (
-        receiver: Receiver | undefined,
         shouldRemoveDrain: "removeDrainHandler" | "leaveDrainHandler"
       ) => void;
 
-      const onSessionError: OnAmqpEvent = (context: EventContext) => {
-        cleanupBeforeResolveOrReject(this._receiver || context.receiver!, "removeDrainHandler");
+      const onError: OnAmqpEvent = (context: EventContext) => {
+        cleanupBeforeResolveOrReject("removeDrainHandler");
+
+        const eventType = context.session?.error != null ? "session_error" : "receiver_error";
+        let error = context.session?.error || context.receiver?.error;
```

Comment:
Are we sure that the `session_error` and `receiver_error` events are always fired with an error?
If they get fired with no error at any time, then in such cases, we fail to log that the event was fired.

We don't need to fix this in this PR as we are only maintaining status quo. 
But, we should have a follow up issue logged.

Same for `session_close` and `receiver_close` events

Reply:
Tracked in #10300